toolsrc/src/vcpkg/sourceparagraph.cpp

```diff
@@ -1271,15 +1271,18 @@ namespace vcpkg
 
                 serialize_optional_array(dep_obj, DependencyDeserializer::FEATURES, features_copy);
                 serialize_optional_string(dep_obj, DependencyDeserializer::PLATFORM, to_string(dep.platform));
-                if (dep.constraint.port_version != 0)
-                {
-                    dep_obj.insert(DependencyDeserializer::PORT_VERSION,
-                                   Json::Value::integer(dep.constraint.port_version));
-                }
-
                 if (dep.constraint.type == Versions::Constraint::Type::Minimum)
                 {
-                    dep_obj.insert(DependencyDeserializer::VERSION_GE, Json::Value::string(dep.constraint.value));
+                    if (dep.constraint.port_version != 0)
+                    {
+                        dep_obj.insert(DependencyDeserializer::VERSION_GE,
+                                       Json::Value::string(
+                                           Strings::concat(dep.constraint.value, '#', dep.constraint.port_version)));
+                    }
+                    else
+                    {
+                        dep_obj.insert(DependencyDeserializer::VERSION_GE, Json::Value::string(dep.constraint.value));
+                    }
```

Comment:
Suggest DRYing this out:

```suggestion
                    string versionConstraintStr;
                    if (dep.constraint.port_version == 0) {
                        versionConstraintStr = dep.constraint.value;
                     } else {
                        versionConstraintStr = Strings::concat(dep.constraint.value, '#', dep.constraint.port_version);
                    }

                    dep_obj.insert(DependencyDeserializer::VERSION_GE, Json::Value::string(std::move(versionConstraintStr)));
```

Reply:
Or even something like:

```
                    string versionConstraintStr = dep.constraint.value;
                    if (dep.constraint.port_version != 0) {
                        versionConstraintStr.push_back('#');
                        versionConstraintStr.append(std::to_string(dep.constraint.port_version));
                    }
```